source/CMakeLists.txt

```
@@ -77,16 +77,55 @@ else()
   message(STATUS "Will not build AMD GPU support")  
 endif (USE_ROCM_TOOLKIT)
 
+#define USE_ASCEND, find tensorflow abi info
+if (USE_ASCEND)
+  add_definitions("-DHUAWEI_ASCEND")
+  set(DP_VARIANT "ascend")
+endif(USE_ASCEND)
+if (USE_ASCEND AND BUILD_CPP_IF)
+  set(PYTHON_EXECUTABLE "python3.7")
```

One may use its own Python executable so it's better to only set when it is not set
```suggestion
  if (NOT DEFINED ${PYTHON_EXECUTABLE})
    set(PYTHON_EXECUTABLE "python3.7")
  endif()
```

---

Good idea, we have fixed that.